src/providers/okta.ts

```diff
@@ -0,0 +1,102 @@
+import { OAuth2Client } from "oslo/oauth2";
+import type { OAuth2ProviderWithPKCE } from "../index.js";
+import { TimeSpan, createDate } from "oslo";
+
+export class Okta implements OAuth2ProviderWithPKCE {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(
+		oktaDomain: string,
+		clientId: string,
+		clientSecret: string,
+		redirectURI: string,
+		authServerId?: string
```

Comment:
Can you make an `option` object instead to be consistent with other providers?

Reply:
Also, I can't find any mention of server IDs?
https://developer.okta.com/docs/guides/implement-grant-type/authcodepkce/main/

Reply:
The server ID is used when not using the built in org authorization server https://developer.okta.com/docs/reference/api/oidc/#composing-your-base-url